web/sharings/sharings.go

```diff
@@ -211,6 +212,51 @@ func RecipientRefusedSharing(c echo.Context) error {
 	return c.Redirect(http.StatusFound, u.String()+"#")
 }
 
+// receiveDocument stores a shared document in the Cozy.
+//
+// If the document to store is a "io.cozy.files" our custom handler will be
+// called, otherwise we will redirect the request to PUT /data/:doctype/:id.
+func receiveDocument(c echo.Context) error {
+	var err error
+	doctype := c.Param("doctype")
+	id := c.Param("id")
+
+	switch c.Param("doctype") {
+	case "":
+		err = echo.NewHTTPError(http.StatusBadRequest,
+			"Missing parameter: doctype")
+	case consts.Files:
+		err = creationWithIDHandler(c)
+	default:
+		// We call PUT /data/:doctype/:docid
```

Comment:
Because I thought it would be best not to change other parts of the code (the function `updateDoc` muste be turned public). If that's ok I'll do it, it is way easier indeed!